test/e2e/v1beta1/scripts/gh-actions/build-load.sh

```diff
@@ -0,0 +1,183 @@
+#!/usr/bin/env bash
+
+# Copyright 2022 The Kubeflow Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This script is used to build all Katib images.
+# Run ./scripts/v1beta1/build.sh <IMAGE_REGISTRY> <TAG> to execute it.
+
+set -o errexit
+set -o pipefail
+set -o nounset
+cd "$(dirname "$0")"
+
+TRIAL_IMAGES=${1:-""}
+CLUSTER_NAME=${2:-""}
+EXPERIMENTS=${3:-""}
+DEPLOY_KATIB_UI=${4:-false}
+
+REGISTRY="docker.io/kubeflowkatib"
+TAG="e2e-test"
+VERSION="v1beta1"
+CMD_PREFIX="cmd"
+SPECIFIED_DEVICE_TYPE_IMAGES=("enas-cnn-cifar10-cpu" "darts-cnn-cifar10-cpu")
+
+IFS="," read -r -a TRIAL_IMAGE_ARRAY <<< "$TRIAL_IMAGES"
+IFS="," read -r -a EXPERIMENT_ARRAY <<< "$EXPERIMENTS"
+
+_build_containers() {
+  CONTAINER_NAME=${1:-"katib-controller"}
+  DOCKERFILE=${2:-"$CMD_PREFIX/katib-controller/$VERSION/Dockerfile"}
+
+  for image in "${SPECIFIED_DEVICE_TYPE_IMAGES[@]}"; do
+    if [ "$image" = "$CONTAINER_NAME" ]; then
+      DOCKERFILE="${DOCKERFILE//-cpu/}"
+      DOCKERFILE="${DOCKERFILE}.cpu"
+      break
+    fi
+  done
+
+  echo -e "\nBuilding $CONTAINER_NAME image with $DOCKERFILE...\n"
+  docker build --platform "$(uname -m)" -t "$REGISTRY/$CONTAINER_NAME:$TAG" -f "../../../../../$DOCKERFILE" ../../../../../
+}
+
+_load_kind_cluster() {
+  CONTAINER_NAME=${1:-"katib-controller"}
+
+  echo -e "\nLoading $CONTAINER_NAME image to $CLUSTER_NAME...\n"
+  kind load docker-image "$REGISTRY/$CONTAINER_NAME:$TAG" --name "$CLUSTER_NAME"
+}
+
+_install_tools() {
+  # install yq
+  if [ -z "$(which yq)" ]; then
+    wget -O /usr/local/bin/yq "https://github.com/mikefarah/yq/releases/download/v4.25.2/yq_$(uname -s)_$(uname -m)"
+    chmod +x /usr/local/bin/yq
+  fi
+}
+
+cleanup_build_cache() {
+  echo -e "\nCleanup Build Cache...\n"
+  docker builder prune
+}
+
+run() {
```

Comment:
If i understand correctly, core images will be built for every job while  suggestion/early stopping images will be built if used in the experiment.
Right? Is there any other way that we can avoid duplicate build? 

Is this https://github.com/docker/build-push-action/blob/master/docs/advanced/share-image-jobs.md helpful in this case?

Reply:
Let me know your thoughts on it. /cc @andreyvelich

Reply:
we can relook at optimising it later.

Reply:
> If i understand correctly, core images will be built for every job while suggestion/early stopping images will be built if used in the experiment.

Yes, that's right. @johnugeorge 

> Is this https://github.com/docker/build-push-action/blob/master/docs/advanced/share-image-jobs.md helpful in this case?

SGTM
I did not know how to do that.
Thanks! @johnugeorge

Reply:
> we can relook at optimising it later. This looks good.

After merging this PR, I would create another PR to optimize the build script.
WDYT? @johnugeorge

Reply:
Sure Thanks @tenzen-y